service/storage_service.cc

```diff
@@ -962,7 +956,7 @@ class storage_service::ip_address_updater: public gms::i_endpoint_state_change_s
                 co_await utils::get_local_injector().inject("ip-change-raft-sync-delay", std::chrono::milliseconds(500));
                 // Set notify_join to true since here we detected address change and drivers have to be notified
                 nodes_to_notify_after_sync nodes_to_notify;
-                co_await _ss.raft_topology_update_ip(id, endpoint, &nodes_to_notify);
+                co_await _ss.raft_topology_update_ip(id, endpoint, co_await _ss.get_host_id_to_ip_map(), &nodes_to_notify);
```

Comment:
This part remains basically unchanged, the old code executes `get_host_id_to_ip_map` at the beginning of `raft_topology_update_ip`. Why are we sure that calling `get_host_id_to_ip_map` here cannot happen during the topology state application? Is this thanks to `hold_read_apply_mutex` above?

Also, the long comment above seems outdated. There is no `sync_raft_topology_nodes` anymore (you removed it in fbfef6b28a533fed28c8e005980634f2ec64906e) and I see for example
```
// To avoid this, we don't wait for sync_raft_topology_nodes to finish.
```
You could update the comment.